index.d.ts

```diff
@@ -0,0 +1,16 @@
+export type Scalar = number | string | boolean | bigint | null | undefined;
```

Comment:
Do we really want to support all of these `code` types?  I would have though it'd just be `number | string`

We do null-guard on `code` in the implementation but maybe that's unnecessary.